src/static/js/getChatUpdate.js

```diff
@@ -0,0 +1,123 @@
+  [{ tokenId: 26, updateOn: null, content: "New Message 20", sender: "Bob" }],
+];
+
+function scrollToBottom() {
+  var element = document.getElementById("messageContainer");
+  element.scrollTop = element.scrollHeight - element.clientHeight;
+}
+
+function getMessages(tokenId) {
+  try {
+    const message = fakeMessage[tokenId];
+    return message;
+  } catch (err) {
+    console.log(err);
+    return null;
+  }
+}
+
+// Function to simulate receiving messages from the backend
+function updateView(tokenId) {
+  // Simulated backend response, replace with actual API call
+  // Here, we assume messages are hardcoded for demonstration purposes
+
+  const messagesFromBackend = getMessages(tokenId);
+  if (messagesFromBackend == null) {
+    console.log("No new messages");
+  } else {
+    const messageContainer = document.getElementById("messageContainer");
+
+    for (let i = 0; i < messagesFromBackend.length; i++) {
+      const message = messagesFromBackend[i];
+      if (message.updateOn != null) {
+        // modify the message in message Container that has the same tokenId as updateOn
+        const messageElement = document.getElementById(message.updateOn);
+        messageElement.textContent = getMessageText(message);
+        messageElement.id = message.tokenId;
+      } else {
+        // create a new message
+        const messageElement = document.createElement("div");
+        messageElement.textContent = getMessageText(message);
+        messageElement.id = message.tokenId;
+        messageElement.style.fontSize = "20px";
+        if (message.sender == name) {
+          messageElement.classList.add("message");
+        }
+        messageContainer.appendChild(messageElement);
+      }
+    }
+
+    // Update the current token ID
+    console.log(messagesFromBackend[messagesFromBackend.length - 1].tokenId);
+    currentTokenId =
+      messagesFromBackend[messagesFromBackend.length - 1].tokenId;
+  }
+}
+
+// Function to format message text for display
+function getMessageText(message) {
+  if (message.content !== null) {
+    return message.sender + ": " + message.content;
+  } else {
+    return (
+      message.sender + ": " + `Message ${message.tokenId} deleted`
+    );
+  }
+}
+
+document
+  .getElementById("messageContainer")
+  .addEventListener("click", function (event) {
+    // Check if the clicked element has the class 'message'
+    if (event.target.classList.contains("message")) {
+      // Apply the effect (you can customize this part)
+      event.target.style.backgroundColor = "#e0e0e0";
+
+      console.log("Element clicked:", event.target.textContent);
+    }
+  });
+
+// Simulate initial message retrieval
+updateView(currentTokenId);
+
+// Simulate periodic message updates (you may use WebSocket or other real-time mechanisms)
+setInterval(() => {
+  console.log("Updating messages...");
+  updateView(currentTokenId);
+  scrollToBottom();
```

Comment:
This function is buggy because it doesn't let me scroll back to the top and remain there. I think you need to end the setInterval once we reach the last message.

Reply:
Opsi. Sorry for this. I just fixed this in the latest commit (also merged with the latest dev branch). This fix the issue -- you will still be taken down to the bottom of the page if you receive new message; If there's no new message, you can definitely stay at the top without any interference